### PR TITLE
Fix infinite loop during zip creation

### PR DESCRIPTION
**Pull request by LukaszKondrackiDynatrace**

### CODE REVIEW

**Review by allgreed — COMMENTED** [review]
wарнинг нeт?
also: if you'd please squash the commits, since they refer to one self-contained change ;)

File: dtcli/scripts/dt.py [hunk 1]
```diff
@@ -552,8 +553,12 @@ def gencerts(**kwargs):
 def build(**kwargs):
     extension_dir_path = kwargs["extension_directory"]
     utils.require_dir_exists(extension_dir_path)
-
     target_dir_path = kwargs["target_directory"]
+
+    if extension_dir_path == target_dir_path or is_subdirectory(extension_dir_path, target_dir_path):
+        click.echo("Warning: extension_directory is the same as or contains target_directory\n"
```
Comment (allgreed) [inline dtcli/scripts/dt.py:559]: I really dislike the errors in the form "a or b is wrong", there are even memes about it. Can we be more precise?

File: dtcli/scripts/dt.py [hunk 2]
```diff
@@ -552,8 +553,12 @@ def gencerts(**kwargs):
...
+                   "All files in target directory will be included in the zip file", err=True)
```
Comment (allgreed) [inline dtcli/scripts/dt.py:560]: No they won't! We can inform about a potential loop that would happen and that we're stopping it

File: dtcli/scripts/dt.py [hunk 3]
```diff
@@ -619,6 +624,10 @@ def assemble(source, destination, force):
     if destination.exists() and not force:
         raise click.BadParameter(f"destination {destination} already exists, please try again with --force to proceed irregardless", param_hint="--source")
 
+    if source == destination or is_subdirectory(source, destination):
```
Comment (allgreed) [inline dtcli/scripts/dt.py:627]: It shouldn't be possible for source to equal destination -> right now you can pass this, but if you flip `dir_okay` to False on the `output` it should not come through

**Review by allgreed — CHANGES_REQUESTED** [review]

File: dtcli/scripts/dt.py (hunk 2 above)
Comment (LukaszKondrackiDynatrace) [reply dtcli/scripts/dt.py:560]: They will except for one file which is exception.zip

**Review by allgreed — APPROVED** [review]

**Pull request merged**